src/components/Dashboard/Header/Authentication/Signup.jsx

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from 'react';
+import {
+  Form, Button, Container, Row, Col,
+} from 'react-bootstrap';
+import { Redirect } from 'react-router-dom';
+import { Toaster, Intent } from '@blueprintjs/core';
+import db from '../../../../fire';
+import NavBar from '../Header';
+
+export default class Signup extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      redirect: false,
+    };
+  }
+
+  createAccount = (event) => {
+    event.preventDefault();
+    const email = this.emailInput.value;
+    const password = this.passwordInput.value;
+
+    if (this.validate(event)) {
+      console.log(event);
+      db.auth().fetchProvidersForEmail(email)
+        .then((providers) => {
+          console.log(providers);
+          if (providers.length === 0) {
+            // catch user
+            return db.auth().createUserWithEmailAndPassword(email, password);
+          } else {
+            this.loginForm.reset();
+            this.triggerValidation(this.emailInput, 'That email is taken. Try another.');
+          }
+        })
+        .then((user) => {
+          if (user && user.user && user.user.email) {
+            this.loginForm.reset();
+            this.setState({ redirect: true });
+          }
+        })
+        .catch((error) => {
+          this.toaster.show({ intent: Intent.DANGER, message: error.message });
+        });
+    }
+  }
+
+  validate = () => {
+    const name = this.nameInput.value;
+    const email = this.emailInput.value;
+    const password = this.passwordInput.value;
+    const confirmPassword = this.confirmPasswordInput.value;
+    let valid = true;
+    if (!name) {
+      this.triggerValidation(this.nameInput, 'Name is required.')
+      valid = false;
+    } else if (!email) {
+      this.triggerValidation(this.emailInput, 'Email is required.')
+      valid = false;
+    } else if (password.length < 6) {
+      this.triggerValidation(this.passwordInput, 'Password must be 6 characters and longer.')
+      valid = false;
+    } else if (password !== confirmPassword) {
+      this.triggerValidation(this.confirmPasswordInput, 'Password does not match.')
+      valid = false;
+    }
+    return valid;
+  }
+
+  triggerValidation = (input, message) => {
+    input.setCustomValidity(message);
+    input.reportValidity();
+  }
+
+  onChange = (event) => {
```

Comment:
maybe better to rename to `onInputChange /onFieldChange`, so it will be clear at once what this method in used for